ci/lib/shard.sh

```diff
@@ -56,6 +56,8 @@ readonly PUBSUB_SHARD=(
 readonly STORAGE_SHARD=(
   storage
   experimental-storage_grpc
+  # universe_domain is included because its deps are already built by storage.
+  universe_domain
```

Comment:
So our only cmake build for this library is in `clang-tidy-storage-pr`. We will want to verify the install side of things (as well as other things like C++20).

I think something like `cmake-single-feature` will break if we tried to `-DGOOGLE_CLOUD_CPP_ENABLE=universe_domain` on its own. I expected to see logic that says REST, gRPC are enabled if "universe_domain" is in the enable list.

This PR is already large-ish. If you want to handle the cmake installing in a follow up that is fine with me.

I wouldn't be surprised if we had to do some extra maneuvering to get Doxygen to work, too.